### DISCUSSION TITLE
"Customize Sources" for variant listing table?

### DISCUSSION THREAD

**Discussion by TheFunkyMonk** [craftcms/commerce#3784] [General]
In the table view for a product's variants, I see the "Use defaults" option on the bottom left:
![2024-11-22 at 12 58 22](https://github.com/user-attachments/assets/97b88625-02ec-496a-8897-0644dabfa9ae)

However, I'm not sure where to actually configure these defaults (i.e. the equivalent of "Customize Sources" in the Entries panel). Am I missing something, or is this not currently possible?

**Comment by lukeholder** [comment]
Those are just the defaults the variant element specifies as the default and are not configurable.

Currently the colunns you turn on additionally are globally the default until you press that button.

We can leave this discussion open for suggestions on how default should be managed for variants. Maybe on a per product type basis?

**Comment by turnstylerj** [comment]
Yeah, it would be helpful to be able to set the defaults on a per-product basis. There are some custom fields we have on variants that are very relevant to the content authors but they always forget the column view can be customized, so it'd be nice to set those fields to show up by default.

**Comment by lukeholder** [comment]
Thanks for the request. We will leave this ticket open while we discuss internally.

Will update this discussion if we have any news on it. Thanks
